docker/nginx-consul-template/nginx.conf.tmpl

```diff
@@ -0,0 +1,32 @@
+events {
+    worker_connections 4096;
+}
+
+http {
+{{- with $service_query := (printf "%s.%s" (env "DE_ENV") (env "NGINX_PROXY_SERVICE_NAME")) }}
+  {{- if service $service_query }}
+    upstream service {
+    {{- range service $service_query }}
+      server {{.Address}}:{{.Port}} max_fails=3 fail_timeout=60 weight=1 {{- if not (.Tags.Contains (key_or_default (printf "%s/%s/color" (env "DE_ENV") (env "NGINX_PROXY_SERVICE_NAME")) "green"))}} backup{{end}}; # {{.ID}} {{.Tags}}
```

Comment:
Two things I'm not sure if I should try to build in or not and I'd love opinions on:

a.) when no color is configured in consul, should I try to have it automatically detect and choose a color that has servers? alternatively, should I make it choose all servers by having it default to `(env "DE_ENV")` (which will always match in this line)

b.) when a color is configured and there are servers, but all for other colors, right now this will generate an invalid upstream in which all the servers are marked 'backup'; then, the config will update but the `nginx -s reload` will fail so it'll keep using the old config. consul-template sadly does not support querying by multiple tags at once in the `service` function it provides, and I thought it'd make sense to have alternate colors listed as backups in any case, but should I put more effort into detecting this case of "you've probably got the wrong color configured"?